### PR TITLE
cephfs-shell: Return proper error code

### PR DESCRIPTION
**Pull request by varshar16**
Fixes: https://tracker.ceph.com/issues/46420
Signed-off-by: Varsha Rao <varao@redhat.com>

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by varshar16** [comment]
Fixed flake8 errors:
```
py3 run-test: commands[0] | flake8 --ignore=W503 --max-line-length=100 cephfs-shell
cephfs-shell:62:9: E126 continuation line over-indented for hanging indent
cephfs-shell:84:9: E123 closing bracket does not match indentation of opening bracket's line
```

**Comment by varshar16** [comment]
jenkins test dashboard backend

**Comment by batrick** [comment]
jenkins test dashboard backend

**Comment by dmick** [comment]
If we're doing such a nonstandard thing as defining our own non-errno exit codes, this really has to have documentation associated

File: src/tools/cephfs/cephfs-shell [hunk 1]
```diff
@@ -57,6 +58,29 @@ except ImportError:
 
 cephfs = None   # holds CephFS Python bindings
 shell = None    # holds instance of class CephFSShell
+exit_codes = {'Misc': 1,
```
Comment (rishabh-d-dave) [inline src/tools/cephfs/cephfs-shell:61]: Won't it be better to align our exit values with the standard one? Setting `EPERM` to 1 `ENOENT` to 2 and so on.

File: src/tools/cephfs/cephfs-shell [hunk 2]
```diff
@@ -1093,8 +1109,7 @@ class CephFSShell(Cmd):
                     poutput("{}:".format(path.decode('utf-8')))
                     print_list(items)
                 except OSError as e:
-                    perror("'{}': {}".format(e.filename, e.strerror))
-                    self.exit_code = e.errno
+                    set_exit_code_msg(e.errno, "'{}': {}".format(e.filename, e.strerror))
```
Comment (rishabh-d-dave) [inline src/tools/cephfs/cephfs-shell:1112]: s/`e.filename`/`e.filename.decode()`.

Comment (rishabh-d-dave) [reply src/tools/cephfs/cephfs-shell:1112]: Also, s/`"'{}': {}".format(e.filename, e.strerror`/`f'"{e.filename.decode()}": {e.strerror}'`

File: src/tools/cephfs/cephfs-shell [hunk 3]
```diff
@@ -1064,8 +1080,8 @@ class CephFSShell(Cmd):
         try:
             os.chdir(os.path.expanduser(args.path))
         except OSError as e:
-            perror("Cannot change to {}: {}".format(e.filename, e.strerror))
-            self.exit_code = e.errno
+            set_exit_code_msg(e.errno, "Cannot change to {}: {}".format(e.filename,
```
Comment (rishabh-d-dave) [inline src/tools/cephfs/cephfs-shell:1083]: s/`e.filename`/`e.filename.decode()`.

**Comment by varshar16** [comment]
> If we're doing such a nonstandard thing as defining our own non-errno exit codes, this really has to have documentation associated

Agreed, I have updated the doc.

File: src/tools/cephfs/cephfs-shell (hunk 3 above)
Comment (varshar16) [reply src/tools/cephfs/cephfs-shell]: Done

File: src/tools/cephfs/cephfs-shell (hunk 2 above)
Comment (varshar16) [reply src/tools/cephfs/cephfs-shell]: done

**Comment by varshar16** [comment]
jenkins render docs

File: src/tools/cephfs/cephfs-shell (hunk 1 above)
Comment (varshar16) [reply src/tools/cephfs/cephfs-shell:61]: imo it is fine as we are defining custom exit values.

Comment (rishabh-d-dave) [reply src/tools/cephfs/cephfs-shell:61]: I don't mind creating custom values, IMO we must align them as far as possible.

**Comment by dparmar18** [comment]
I was working with some of the cephfs-shell related trackers and came across `set_exit_code_msg()`, it's a good addition to cephfs-shell but I feel like the exit-codes should be aligned with the existing ones(errno.py). It maintains a dictionary where the first two keys are `str` but the rest are not. It looks a bit non-uniform. Maybe we can try keep the errnos aligned to existing ones?

**Pull request merged**